server/webpa.go

```diff
@@ -220,11 +225,13 @@ type WebPA struct {
 // create one using Health.NewHealth.  In either case, if Health.Address is not supplied, no health server
 // will be instantiated.
 //
+// The caller may also pass a gatherer type. If it is not provided, the default provided by prometheus is used.
+//
 // The supplied http.Handler is used for the primary server.  If the alternate server has an address,
-// it will also be used for that server.  The health server uses an internally create handler, while the pprof
-// server uses http.DefaultServeMux.  The health Monitor created from configuration is returned so that other
+// it will also be used for that server.  The health server uses an internally create handler, while pprof and metrics
+// servers use http.DefaultServeMux.  The health Monitor created from configuration is returned so that other
 // infrastructure can make use of it.
-func (w *WebPA) Prepare(logger log.Logger, health *health.Health, primaryHandler http.Handler) (health.Monitor, concurrent.Runnable) {
+func (w *WebPA) Prepare(logger log.Logger, health *health.Health, primaryHandler http.Handler, gatherer prometheus.Gatherer) (health.Monitor, concurrent.Runnable) {
```

Comment:
Move the gatherer into the WebPA struct.  That lets client code more easily specify the default by simply not setting that struct member.  Eventually, we'll remove the health parameter as well.